src/pipelines/aws/aws-pipeline.ts

```diff
@@ -158,11 +158,20 @@ export default class AWSPipeline implements Pipeline {
       '$HEIGHT',
       targetResolution.height.toString()
     );
-    settingsStr = this.stringReplacement(
-      settingsStr,
-      '$BITRATE',
-      targetBitrate.toString()
-    );
+    if (settingsStr.includes('$BITRATE')) {
+      settingsStr = this.stringReplacement(
+        settingsStr,
+        '$BITRATE',
+        targetBitrate.toString()
+      );
+    }
+    if (settingsStr.includes('$QVBRQUALITYLEVEL')) {
```

Comment:
I think it would be better if we allow substituing arbitraty 'variables' in the template by defining values in the `pipelineVariables` property, similar to this example: https://github.com/eyevinn/autovmaf?tab=readme-ov-file#using-variables-in-the-job-definition .